documentation/sphinx/source/auditstorage.rst

```diff
@@ -0,0 +1,86 @@
+The tool also checks if the ServerKeys are consistent to StorageServer (SS) location shard mapping.
+Note that ServerKeys is a part of the system key space while the SS location shard mapping is stored in the storage server locally.
+The tool checks a range is assigned to the SS in ServerKeys if and only if the range is owned by the SS in the local shard mapping.
+
+The AuditStorage tool has following features:
+
+* End-to-end completeness check --- The checker continues until all ranges are marked as complete. AuditStorage persists progress to a metadata. This is a major difference from the Consistency Checker Urgent tool.
+* Scalability --- Adding more parallelism results in nearly linear speedup. This is controlled by the knob CONCURRENT_AUDIT_TASK_COUNT_MAX.
+* Progress monitoring --- Providing FDBCLI command to check the progress of the checking.
+* Fault tolerance --- Failures do not impact the checker process. Checking failures are automatically retried.
+* Simple to setup --- Different from the Consistency Checker Urgent tool, the AuditStorage tool uses DD and SSes to conduct checking. Therefore, a cluster can run the AuditStorage tool without any additional setup.
+
+How to use?
+-----------
+* Start new audit job: audit_storage [replica\|locationmetadata\|ssshard] [BeginKey] [EndKey] (For example, to check all replicas, we can run audit_storage replica "" \\xff\\xff)
+* List recent jobs: get_audit_status [replica\|locationmetadata\|ssshard] recent
+* Check a job progress: get_audit_status [replica\|locationmetadata\|ssshard] progress [AuditID]
+* Cancel an audit: audit_storage cancel [replica\|locationmetadata\|ssshard] [AuditID]
+
+There three auditTypes:
+
+1. replica: This audit checks the consistency of user data between replicas of all DCs. SSAuditStorageShardReplicaError trace event is generated if any inconsistency is detected.
+2. locationmetadata: This audit checks the consistency between KeyServer and ServerKey space. DDDoAuditLocationMetadataError trace event is generated if any inconsistency is detected.
+3. ssshard: This audit checks the consistency between KeyServer and storage server in-memory shard information. SSAuditStorageSsShardError trace event is generated if any inconsistency is detected.
+
+BeginKey and EndKey decide the scope of the consistency check of user data (replica). Note that the locationmetadata and ssshard always check the consistency on all key space no matter the user input. 
+
+Compared to Consistency Checker Urgent
+--------------------------------------
+
+AuditStorage offers significant improvements over the existing consistency checker urgent in several key areas:
+
+* ConsistencyCheckerUrgent is not efficient because it does not persist progress to the metadata. On the other hand, AuditStorage persists progress to the metadata. As a result, the AuditStorage tool improves the efficiency by avoiding doing repeated work.
```

Comment:
```suggestion
* ConsistencyCheckerUrgent is not efficient because it does not persist progress to the database. On the other hand, AuditStorage persists progress to the metadata stored in the database. As a result, the AuditStorage tool improves the efficiency by avoiding doing repeated work.
```